src/puzzle/Piece.js

```diff
@@ -158,7 +158,9 @@ pzpr.classmgr.makeCommon({
 			}
 
 			this.addOpe(prop, this[prop], num);
-			this[prop] = num;
+			if (!Object.isFrozen(this)) {
+				this[prop] = num;
+			}
```

Comment:
I guess these are the scary upates to `emptycell` etc you were talking about. I should look into this a bit more. E.g. `addOpe` makes no sense if we ignore the update after (which was then already the case as far as I understand).

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode

> [...] Any assignment that **silently fails** in normal code (assignment to a non-writable global or property, assignment to a getter-only property, assignment to a new property on a non-extensible object) will throw in strict mode [...]

(emphasis mine)

Reply:
In understand that this is not a functional change. But the explicit `isFrozen` test somehow "normalizes" the current possibly buggy behaviour.

Reply:
Anyway, I'll look into this separately.